File: Week-1-April-1st-7th/Counting-elements/counting-elements.cpp
// Time complexity: O(n) .... Space complexity: O(n)
class Solution {
public:
    int countElements(vector<int>& arr) {
        unordered_set<int> s;
        int ans = 0;
        for(auto a : arr){
            s.insert(a); //O(1)
        }
        for(auto a: arr){
            ans += 1 ? s.count(a + 1) == 1 : 0;
        }
        return ans;
    }
};

File: Week-1-April-1st-7th/Counting-elements/counting-elements.py
# Time complexity: O(n^2).... Space complexity: O(1)
class Solution:
    def countElements(self, arr: List[int]) -> int:
        ans = 0
        for x in arr:
            if (x + 1) in arr:
                ans += 1
        return ans

# Time complexity: O(n)... Space complexity: O(n)
#Note: "in" operator in Python has time complexity is: O(1)
class Solution:
    def countElements(self, arr: List[int]) -> int:
        tmp = set() # set
        for a in arr:
            tmp.add(a)
        return sum([1 if (a + 1) in tmp else 0 for a in arr])